ksqldb-engine/src/test/java/io/confluent/ksql/function/udaf/offset/EarliestByOffsetUdafTest.java

```diff
@@ -73,6 +111,28 @@ public void shouldMerge() {
     assertThat(merged1, is(agg1));
     assertThat(merged2, is(agg1));
   }
+  
+  @Test
+  public void shouldMerge2Integers() {
+    // Given:
+    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestInteger(2);
+
+    Struct struct1 = EarliestByOffset.createStruct(STRUCT_INTEGER, 123);
+    Struct struct2 = EarliestByOffset.createStruct(STRUCT_INTEGER, 321);
+    List<Struct> agg1 = new ArrayList<>(Arrays.asList(struct1, struct2));
+    
+    Struct struct3 = EarliestByOffset.createStruct(STRUCT_INTEGER, 543);
+    Struct struct4 = EarliestByOffset.createStruct(STRUCT_INTEGER, 654);
+    List<Struct> agg2 = new ArrayList<>(Arrays.asList(struct3, struct4));
+    
+    // When:
+    List<Struct> merged1 = udaf.merge(agg1, agg2);
+    List<Struct> merged2 = udaf.merge(agg2, agg1);
+
+    // Then:
+    assertThat(merged1, is(agg1));
+    assertThat(merged2, is(agg1));
+  }
```

Comment:
A better test would be if the structs in each parameter had interleaved sequence numbers, as this ensures the sort is happening:

```suggestion
  @Test
  public void shouldMergeIntegers() {
    // Given:
    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestInteger(2);

    final Struct struct1 = EarliestByOffset.createStruct(STRUCT_INTEGER, 123);
    final Struct struct2 = EarliestByOffset.createStruct(STRUCT_INTEGER, 321);
    final Struct struct3 = EarliestByOffset.createStruct(STRUCT_INTEGER, 543);
    final Struct struct4 = EarliestByOffset.createStruct(STRUCT_INTEGER, 654);
    final List<Struct> agg1 = new ArrayList<>(Arrays.asList(struct1, struct4));
    final List<Struct> agg2 = new ArrayList<>(Arrays.asList(struct2, struct3));
    
    // When:
    final List<Struct> merged1 = udaf.merge(agg1, agg2);
    final List<Struct> merged2 = udaf.merge(agg2, agg1);

    // Then:
    assertThat(merged1, contains(struct1, struct2));
    assertThat(merged2, contains(struct1, struct2));
  }
```